cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_adapters.go

```diff
@@ -134,3 +136,51 @@ func (s *Server) getAvailablePackageDetail(pkgMetadata *datapackagingv1alpha1.Pa
 
 	return availablePackageDetail, nil
 }
+
+func (s *Server) getInstalledPackageSummary(pkgInstall *packagingv1alpha1.PackageInstall, pkgMetadata *datapackagingv1alpha1.PackageMetadata, pkgVersionsMap map[string][]pkgSemver, cluster string) (*corev1.InstalledPackageSummary, error) {
+	// get the versions associated with the package
+	versions := pkgVersionsMap[pkgInstall.Spec.PackageRef.RefName]
+	if len(versions) == 0 {
+		return nil, fmt.Errorf("no package versions for the package %q", pkgInstall.Spec.PackageRef.RefName)
+	}
+
+	// Carvel uses base64-encoded SVG data for IconSVGBase64, whereas we need
+	// a url, so convert to a data-url.
+	iconUrl := ""
+	if pkgMetadata.Spec.IconSVGBase64 != "" {
+		iconUrl = fmt.Sprintf("data:image/svg+xml;base64,%s", pkgMetadata.Spec.IconSVGBase64)
```

Comment:
I wonder if this is one of the things causing the slowness... some of the SVG's are quite big (300k) and we're having to copy them to each and return via the network (rather than a URL to a resource). If so, definitely a place where caching may help us.

In the mean-time, it might be worth replacing the `fmt.Sprintf` here with a https://pkg.go.dev/strings#Builder so that we're not copying the string when assigned below. Instead we can just using `builder.String()` in the final struct below?

Reply:
Another thought: if we find that it is the collection of all the svg data being fetched across the network for each installed package detail and sent all together in the one response that is the issue, we could even add an extra endpoint just to this kapp-controller plugin for `AvailablePackageIcon`, expose it via the gateway, then populate the IconURL of the summary request with the gateway endpoint. This would mean that the response to `GetInstalledPackageSummaries` (and `GetAvailablePackageSummaries`) would just have a normal URL for the icon which would be fetched separately by the browser.

We'd still want to avoid pulling the icon data over the network from the k8s server when getting the metadata for other fields though. Have to see if that's possible.